torch/_higher_order_ops/invoke_subgraph.py

```diff
@@ -63,9 +67,34 @@ def __call__(
 invoke_subgraph = InvokeSubgraphHOP()
 
 
-def invoke_subgraph_placeholder(subgraph, *args, **kwargs):
-    # Just a placeholder for Dynamo to replace with invoke_subgraph
-    return subgraph(*args, **kwargs)
+def invoke_subgraph_placeholder(func, *args, **kwargs):
+    if torch.compiler.is_dynamo_compiling():
+        # This is just a placeholder for Dynamo to replace with invoke_subgraph
+        raise RuntimeError("invoke_subgraph should not be called directly in Dynamo")
+
+    if torch.compiler.is_compiling():
+        # For non-strict export tracing, we still want to go through Dynamo
+        from torch._dynamo.backends.debugging import (
+            make_eager_backend_with_torch_function_mode,
+        )
+
+        def _invoke_subgraph_placeholder_wrapper(func, args):
+            return invoke_subgraph_placeholder(func, *args)
+
+        with _set_compilation_env(), torch._dynamo.utils.disable_cache_limit(), _temp_remove_pre_dispatch_torch_function_mode():
+            with _temp_remove_metadata_torch_function_mode() as metadata_mode:
+                if metadata_mode:
+                    backend = make_eager_backend_with_torch_function_mode(metadata_mode)
+                else:
+                    backend = "eager"
+
+                return torch.compile(
```

Comment:
Why do you need a hop_dispatch_key_set in proxy_tensor.py if you're going through Dynamo?

Reply:
I believe this helps us generate the TorchIR graph, but once we get to ProxyDispatch tracing, since export doesn't create a TracingContext, it doesn't have anywhere to store the AtenIR graphs.

Reply:
where does the torch.compile path store the hop_dispatch_key_set? It would be nice if both paths end up storing things in the same place (and it's OK if that place is proxy_tensor, but I am trying to figure out what's going on)

Reply:
torch.compile stores it in the TracingContext. Maybe it would be more correct for export to create a TracingContext?